include/zephyr/pm/policy.h

```diff
@@ -86,6 +86,51 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks);
 /** Special value for 'all substates'. */
 #define PM_ALL_SUBSTATES (UINT8_MAX)
 
+/**
+ * @brief Build a pm_policy_state_lock_ function for a node property's
+ *	  power-state phandle index.
+ *
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name containing the
+ *	       power-states phandle
+ * @param idx phandles item index
+ * @param fn pm_policy_state_lock_ function to call with the herein
+ *	     generated parameters (enum pm_state, uint8_t substate_id)
+ *
+ * @see PM_POLICY_DT_DRV_INST_LOCK
+ */
+#define PM_POLICY_DT_REINIT_LOCK_FN(node_id, prop, idx, fn)		\
+fn(									\
+	UTIL_CAT(PM_STATE_, DT_STRING_UPPER_TOKEN(			\
+				DT_PHANDLE_BY_IDX(node_id, prop, idx),	\
+				power_state_name)),			\
+	DT_PROP_BY_PHANDLE_IDX_OR(node_id, prop, idx,			\
+				  substate_id, 0))
+
+/**
+ * @brief Build a series of pm_policy_state_lock_put functions for each of a
+ *	  node's reinit-power-states.
+ *
+ * @see pm_policy_state_lock_put()
+ */
+#define PM_POLICY_INST_REINIT_LOCK					\
+	DT_FOREACH_PROP_ELEM_SEP_VARGS(					\
+			DT_DRV_INST(0), reinit_power_states,		\
```

Comment:
This is obviously problematic, we need a method to apply this to the correct instance.

Reply:
Solved via more central approach to the reinit state instance handling via driver config struct.